### PR TITLE
MRG: Make inv["noise_cov"] a Covariance

### PR DESCRIPTION
**Pull request by Eric89GXL**
Some refactoring to clean up `Covariance` instantiation and allow `inv['noise_cov']` to be a `Covariance` instance. cc @nfoti

Ready for review/merge from my end.

### CODE REVIEW

File: mne/cov.py [hunk 1]
```diff
@@ -290,12 +322,8 @@ def make_ad_hoc_cov(info, verbose=None):
...
                              (grad_std, mag_std, eeg_std)):
         data[pick_types(info, meg=meg, eeg=eeg)] = val * val
-    cov = Covariance(None)
-    cov.update(kind=FIFF.FIFFV_MNE_NOISE_COV, diag=True, dim=len(data),
-               names=info['ch_names'], data=data, projs=info['projs'],
-               bads=info['bads'], nfree=0, eig=None, eigvec=None,
-               info=info)
-    return cov
+    return Covariance(data, info['ch_names'], info['bads'], info['projs'],
+                      nfree=0)
```
Comment (agramfort) [inline mne/cov.py]: you're not setting diag, dim, kind and info. Any consequence?

File: mne/cov.py [hunk 2]
```diff
@@ -723,11 +741,8 @@ def _unpack_epochs(epochs):
 
     covs = list()
     for this_method, data in cov_data.items():
-        cov = Covariance(None)
-        cov.update(kind=1, diag=False, dim=len(data['data']), names=ch_names,
-                   data=data.pop('data'), projs=projs, bads=info['bads'],
-                   nfree=n_samples_tot, eig=None, eigvec=None)
-
+        cov = Covariance(data.pop('data'), ch_names, info['bads'], projs,
+                         nfree=n_samples_tot)
```
Comment (agramfort) [inline mne/cov.py]: kind? dim?

**Comment by agramfort** [comment]
that's it

thanks for cleaning up my mess :)

File: mne/cov.py [hunk 3]
```diff
@@ -76,17 +96,26 @@ class Covariance(dict):
...
-            return
-
-        # Reading
-        fid, tree, _ = fiff_open(fname)
-        self.update(_read_cov(fid, tree, FIFF.FIFFV_MNE_NOISE_COV))
-        fid.close()
+        diag = True if data.ndim == 1 else False
+        self.update(data=data, dim=len(data), names=names, bads=bads,
+                    nfree=nfree, eig=eig, eigvec=eigvec, diag=diag,
+                    projs=projs, kind=FIFF.FIFFV_MNE_NOISE_COV)
```
Comment (Eric89GXL) [inline mne/cov.py]: @agramfort those get set automatically here, so no consequence (other than cleaner/DRY code) to not setting them where you mention

Comment (Eric89GXL) [inline mne/cov.py]: (and you didn't notice that `diag` was no longer being explicitly set, either :) )

Comment (agramfort) [inline mne/cov.py]: ok you win

@dengemann wanna have a look?

**Comment by dengemann** [comment]
Thanks Eric!

**Pull request merged**